### PR TITLE
New fipname option

### LINKED ISSUE #370: [prtvol2csv] Support for more than FIPNUM
User request to support different FIPXXXX, not only FIPNUM. Underlying ecl2df fipreports already supports this.

Output must be set to at least `FIP=3` for Eclipse to write RESERVOIR VOLUMES table for more than FIPNUM.

### PR DESCRIPTION
**Pull request by lilbe66**
Resolves #370 .  Added an new option in prtvol2csv (--fipname) to be able to extract volumes from the Eclipse PRT file for a selected FIP-vector.

### CODE REVIEW

File: docs/scripts/prtvol2csv.rst [hunk 1]
```diff
@@ -39,8 +39,8 @@ Additionally, if the Eclipse DATA file includes::
   RPTSOL
     FIP=2 'FIPRESV' /
 
-the PRT-file will also contain a table with pore volumes pr. phase and pr.
-FIPNUM, which will be added to the exported table.
+the PRT-file will also contain a table with pore volumes pr. phase and pr. FIPNUM,
```
Comment (mferrera) [inline docs/scripts/prtvol2csv.rst:42]:
```suggestion
The PRT-file will also contain a table with pore volumes pr. phase and pr. FIPNUM, 
```

maybe uppercase this while here

File: src/subscript/prtvol2csv/prtvol2csv.py [hunk 2]
```diff
@@ -233,9 +241,14 @@ def reservoir_volumes_from_prt(prt_file: str) -> pd.DataFrame:
     table_found = (
         False  # State determining if current line is in our interesting table or not.
     )
+    fipname_found = fipname == "FIPNUM"  # found the corrent fipname, FIPNUM OK
+
     with Path(prt_file).open(encoding="utf8") as f_handle:
         for line in f_handle:
-            if start_matcher.search(line) is not None:
+            if f"{fipname}  REPORT REGION" in line:
```
Comment (mferrera) [inline src/subscript/prtvol2csv/prtvol2csv.py:248]: Is it possible there could be instances of just a single space, or more than two?

Comment (mferrera) [reply src/subscript/prtvol2csv/prtvol2csv.py:248]:
```python
if fipname in line and "REPORT REGION" in line:
     ...
```

One possible other option -- maybe not the best.

File: src/subscript/prtvol2csv/prtvol2csv.py [hunk 3]
```diff
@@ -264,10 +277,12 @@ def reservoir_volumes_from_prt(prt_file: str) -> pd.DataFrame:
 
     if not records:
         logger.warning("No RESERVOIR VOLUMES table found in PRT file %s", prt_file)
-        logger.warning("Include RPTSOL <newline> FIP=2 'FIPRESV' in Eclipse DATA file")
+        logger.warning(
+            "Include RPTSOL <newline> FIP=2 (or 3)'FIPRESV' in Eclipse DATA file"
```
Comment (mferrera) [inline src/subscript/prtvol2csv/prtvol2csv.py:281]:
```suggestion
            "Include RPTSOL <newline> FIP=2 (or 3) 'FIPRESV' in Eclipse DATA file"
```

File: src/subscript/prtvol2csv/prtvol2csv.py [hunk 4]
```diff
@@ -317,18 +337,28 @@ def prtvol2df(
     simvolumes_df: pd.DataFrame,
     resvolumes_df: pd.DataFrame,
     fipmapper: Optional[FipMapper] = None,
+    fipname: str = "FIPNUM",
 ) -> pd.DataFrame:
     """
     Concatenate two dataframes (with common index) horizontally,
-    and add REGION and ZONE parameter.
+    and if fipname=FIPNUM, add REGION and ZONE parameter.
```
Comment (mferrera) [inline src/subscript/prtvol2csv/prtvol2csv.py:344]:
```suggestion
    and if fipname="FIPNUM", add REGION and ZONE parameter.
```

File: tests/test_prtvol2csv.py [hunk 5]
```diff
@@ -139,6 +140,355 @@ def test_prtvol2csv(tmp_path, mocker):
...
+    # Test for FIPNUM:
+    mocker.patch(
+        "sys.argv", ["prtvol2csv", "--fipname", "FIPNUM", "--debug", str(prtfile)]
+    )
+    prtvol2csv.main()
+    dframe = pd.read_csv("share/results/volumes/simulator_volume_fipnum.csv")
+
+    expected = pd.DataFrame.from_dict(
+        {
+            "FIPNUM": {
```
Comment (mferrera) [inline tests/test_prtvol2csv.py:190]: Is checking a smaller sample of these values sufficient to validate?

**Review by mferrera — COMMENTED** [review]
In general looks good to me. Some tiny comments for your consideration. Roger or Alif should probably be the ones to approve this.

File: docs/scripts/prtvol2csv.rst (hunk 1 above)
Comment (lilbe66) [reply docs/scripts/prtvol2csv.rst:42]: l. 42 is probably intended to be next part of the sentence on line 37, so maybe better to remove the ":" on line 37?  (Upper case after ":")  ("Additionally, if the Eclipse DATA file includes"  < box> "the PRT-file will also ..."  I will try to change like that.

File: src/subscript/prtvol2csv/prtvol2csv.py (hunk 2 above)
Comment (lilbe66) [reply src/subscript/prtvol2csv/prtvol2csv.py]: I looked for another line in the PRT file for the same check, (a few lines above in the PRT file, at start of the report, "  BALZON"  for FIP-keyword FIPZON).  Seems quite robust, according to the Eclipse reference manual (see FIP-keyword).   Not sure how things will work with output from OPM Flow...

File: tests/test_prtvol2csv.py (hunk 5 above)
Comment (lilbe66) [reply tests/test_prtvol2csv.py:190]: Will discuss with Alif

**Review by alifbe — APPROVED** [review]
LGTM 👍

**Comment by lilbe66** [comment]
Merged in PR #748

**Pull request closed without merging**